### PR TITLE
Remove old scheduler configs

### PR DESCRIPTION
**Pull request by tfussell**
Towards https://github.com/giantswarm/giantswarm/issues/12349. Kubernetes 1.16 adds strict decoding of the scheduler config which surfaced that we are setting some fields removed in Kubernetes 1.13. This leads to a warning and slightly less safe config loading. I'm deleting the deleted fields and those where the value matches the default.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: files/config/scheduler.yaml [hunk 1]
```diff
@@ -1,8 +1,4 @@
-kind: KubeSchedulerConfiguration
-algorithmSource:
-  provider: DefaultProvider
 apiVersion: kubescheduler.config.k8s.io/v1alpha1
+kind: KubeSchedulerConfiguration
 clientConnection:
   kubeconfig: /etc/kubernetes/kubeconfig/scheduler.yaml
-failureDomains: kubernetes.io/hostname,failure-domain.beta.kubernetes.io/zone,failure-domain.beta.kubernetes.io/region
-hardPodAffinitySymmetricWeight: 1
```
Comment (tfussell) [inline files/config/scheduler.yaml:8]: Moved to scheduler policy config in https://github.com/kubernetes/kubernetes/pull/44159

File: files/config/scheduler.yaml [hunk 2]
```diff
@@ -1,8 +1,4 @@
-kind: KubeSchedulerConfiguration
-algorithmSource:
-  provider: DefaultProvider
 apiVersion: kubescheduler.config.k8s.io/v1alpha1
+kind: KubeSchedulerConfiguration
 clientConnection:
   kubeconfig: /etc/kubernetes/kubeconfig/scheduler.yaml
-failureDomains: kubernetes.io/hostname,failure-domain.beta.kubernetes.io/zone,failure-domain.beta.kubernetes.io/region
```
Comment (tfussell) [inline files/config/scheduler.yaml:7]: Removed in k8s 1.13 in change from `componentconfig/v1alpha1` to `kubescheduler.config.k8s.io/v1alpha1`.

File: files/config/scheduler.yaml [hunk 3]
```diff
@@ -1,8 +1,4 @@
-kind: KubeSchedulerConfiguration
-algorithmSource:
```
Comment (tfussell) [inline files/config/scheduler.yaml:2]: Deprecated, defaults to `DefaultProvider`.

**Comment by tfussell** [comment]
Tested on `geckon`. Before change:
```
I1124 12:16:58.250247       1 serving.go:312] Generated self-signed cert in-memory
W1124 12:16:58.255259       1 configfile.go:65] using lenient decoding as strict decoding failed: strict decoder error for kind: KubeSchedulerConfiguration
algorithmSource:
  provider: DefaultProvider
apiVersion: kubescheduler.config.k8s.io/v1alpha1
clientConnection:
  kubeconfig: /etc/kubernetes/kubeconfig/scheduler.yaml
failureDomains: kubernetes.io/hostname,failure-domain.beta.kubernetes.io/zone,failure-domain.beta.kubernetes.io/region
hardPodAffinitySymmetricWeight: 1
: v1alpha1.KubeSchedulerConfiguration.HardPodAffinitySymmetricWeight: ReadObject: found unknown field: failureDomains, error found in #10 byte of ...|reDomains":"kubernet|..., bigger context ...|netes/kubeconfig/scheduler.yaml"},"failureDomains":"kubernetes.io/hostname,failure-domain.beta.kuber|...
W1124 12:16:58.866815       1 authentication.go:267] No authentication-kubeconfig provided in order to lookup client-ca-file in configmap/extension-apiserver-authentication in kube-system, so client certificate authentication won't work.
W1124 12:16:58.872418       1 authentication.go:291] No authentication-kubeconfig provided in order to lookup requestheader-client-ca-file in configmap/extension-apiserver-authentication in kube-system, so request-header client certificate authentication won't work.
```
after change:
```
I1124 14:33:02.274304       1 flags.go:33] FLAG: --write-config-to=""
I1124 14:33:03.554689       1 serving.go:312] Generated self-signed cert in-memory
W1124 14:33:05.327813       1 authentication.go:267] No authentication-kubeconfig provided in order to lookup client-ca-file in configmap/extension-apiserver-authentication in kube-system, so client certificate authentication won't work.
W1124 14:33:05.327884       1 authentication.go:291] No authentication-kubeconfig provided in order to lookup requestheader-client-ca-file in configmap/extension-apiserver-authentication in kube-system, so request-header client certificate authentication won't work.
```

**Review by MarcelMue — APPROVED** [review]

**Review by calvix — APPROVED** [review]
lgtm

**Comment by yulianedyalkova** [comment]
Did we made sure that the behaviour that those flags were defining is preserved (by other flags/options)?

**Comment by tfussell** [comment]
> Did we made sure that the behaviour that those flags were defining is preserved (by other flags/options)?

I think we can tell from the code and the warning message (`found unknown field: failureDomains, error found in #10 byte of ...|reDomains":"kubernet`) that the config here wasn't being used. It's worth looking into whether we should be setting this  on a global or pod-specific level. I think the idea is to use the built-in `requiredDuringSchedulingIgnoredDuringExecution` option for `nodeAffinity` on pods when needed.

**Comment by yulianedyalkova** [comment]
I'm not questioning this PR, I'm just assuming we added those flags for a reason in the first place. And it would be nice to eventually evaluate if we still care about them and if there are alternative ways to ensure the behaviour stays the same :) I have no idea what the flags are doing though so I could be talking nonsense 🤷‍♀️

**Comment by tfussell** [comment]
> I'm not questioning this PR, I'm just assuming we added those flags for a reason in the first place. And it would be nice to eventually evaluate if we still care about them and if there are alternative ways to ensure the behaviour stays the same :) I have no idea what the flags are doing though so I could be talking nonsense 🤷‍♀️

For sure. These were deprecated several years ago so it's hard to tell what they did exactly. I tried reading through the Kubernetes git history, but they don't talk about why, only what. My guess is that the scheduler used to handle failure domains in a special way from other affinity/anti-affinity (if affinity even existed back then) but it has been subsumed by new features.

**Pull request merged**